azure-mgmt-batch/src/main/java/com/microsoft/azure/management/batch/implementation/PoolImpl.java

```diff
@@ -0,0 +1,132 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for
+ * license information.
+ */
+
+package com.microsoft.azure.management.batch.implementation;
+
+import com.microsoft.azure.management.batch.BatchAccount;
+import com.microsoft.azure.management.batch.MountConfiguration;
+import com.microsoft.azure.management.batch.NetworkConfiguration;
+import com.microsoft.azure.management.batch.Pool;
+import com.microsoft.azure.management.resources.fluentcore.arm.models.implementation.ExternalChildResourceImpl;
+import rx.Observable;
+import rx.functions.Func1;
+
+import java.util.List;
+
+/**
+ * Implementation for BatchAccount Pool and its parent interfaces.
+ */
+public class PoolImpl
+        extends ExternalChildResourceImpl<Pool,
+        PoolInner,
+        BatchAccountImpl,
+        BatchAccount>
+        implements  Pool,
+        Pool.Definition<BatchAccount.DefinitionStages.WithPool>,
+        Pool.UpdateDefinition<BatchAccount.Update>,
+        Pool.Update {
+
+    protected PoolImpl(
+            String name,
+            BatchAccountImpl batchAccount,
+            PoolInner inner){
+        super(name, batchAccount, inner);
+    }
+
+    protected static PoolImpl newPool(
+            String name,
+            BatchAccountImpl parent) {
+        PoolInner inner = new PoolInner();
+        inner.withDisplayName(name);
+        PoolImpl poolImpl = new PoolImpl(name, parent, inner);
+        return poolImpl;
+    }
+
+    @Override
+    public Observable<Pool> createResourceAsync() {
+        final PoolImpl self = this;
+
+        return this.parent().manager().inner().pools().createAsync(
+                this.parent().resourceGroupName(),
+                this.parent().name(),
+                this.name(), this.inner()).
+                map(new Func1<PoolInner, Pool>() {
+                    @Override
+                    public Pool call(PoolInner inner) {
+                        self.setInner(inner);
+                        return self;
+                    }
+                });
+    }
+
+    @Override
+    public Observable<Pool> updateResourceAsync() {

```

Comment:
Is it same as createResourceAsync? If so I think this could be skipped and use base class impl.

Reply:
We have to override here as they are declared as abstract in base class impl.

Reply:
We have to override here as they are declared as abstract in base class impl.

Reply:
See the @Override here? It is not abstract.

Reply:
Then just call the createResourceAsync (or let them both call private method). Point is to reduce duplicate code if possible.

Reply:
Oh, it's copy-paste issue here for internal call. I will change to call updateAsync.